### ISSUE TITLE
[Improvement]: Provide a quick-fix code action for invalid transfer out of a lock statement

### ISSUE THREAD

**Issue by nipunayf** [ballerina-platform/ballerina-lang#42333]
### Description

For the diagnostic `invalid attempt to transfer out a value from a 'lock' statement with restricted variable usage: expected an isolated expression`, there are common fixes in which you can either return a clone of the selected value or make the respective data storage immutable. All these CAs should be provided to the user so that they can make the decision based on their use case.

### Describe your solution(s)

Consider the following source code with the respective diagnostic.

```ballerina
isolated map<map<string>> mp = {};

isolated function isolatedFn() returns map<string> {
    lock {
        return mp.get("a");
    }
}
```
**FIX1**: Make the data structure immutable, assuming there are no updates to the storage.

```ballerina
isolated map<map<string>> & readonly mp = {}; // changed

isolated function isolatedFn() returns map<string> {
    lock {
        return mp.get("a");
    }
}
```

**FIX2**: Clone the value expression, and let the caller manipulate the cloned value.

```ballerina
isolated map<map<string>> mp = {};

isolated function isolatedFn() returns map<string> {
    lock {
        return mp.get("a").clone(); // changed
    }
}
```

**FIX3**: Clone the value expression, and disallow the caller to manipulate the cloned value.
```ballerina
isolated map<map<string>> mp = {};

isolated function isolatedFn() returns map<string> {
    lock {
        return mp.get("a").cloneReadonly(); // changed
    }
}
```

### Related area

-> Compilation

### Related issue(s) (optional)

https://github.com/ballerina-platform/ballerina-lang/issues/28681

**Comment by MaryamZi** [comment]
Re: FIX1, the entire container doesn't have to be immutable, just the member. That may be a better suggestion in that case.

```ballerina
isolated map<map<string> & readonly> mp = {}; // the map itself is still mutable, only the members should be immutable

isolated function isolatedFn() returns map<string> {
    lock {
        return mp.get("a");
    }
}
```

**Comment by nipunayf** [comment]
The Fix#1 is blocked by https://github.com/ballerina-platform/ballerina-lang/issues/42725
